pkg/util/utils_supported.go

```diff
@@ -60,6 +60,36 @@ func GetRootlessRuntimeDir() (string, error) {
 	return rootlessRuntimeDir, nil
 }
 
+// GetRootlessConfigHomeDir returns the config home directory when running as non root
+func GetRootlessConfigHomeDir() (string, error) {
+	var rootlessConfigHomeDirError error
+
+	rootlessConfigHomeDirOnce.Do(func() {
+		home := os.Getenv("HOME")
+		resolvedHome, err := filepath.EvalSymlinks(home)
+		if err != nil {
+			rootlessConfigHomeDirError = errors.Wrapf(err, "cannot resolve %s", home)
+			return
+		}
+
+		cfgHomeDir := os.Getenv("XDG_CONFIG_HOME")
+		if cfgHomeDir == "" {
+			tmpDir := filepath.Join(resolvedHome, ".config")
+			os.MkdirAll(tmpDir, 0700)
+			st, err := os.Stat(tmpDir)
+			if err == nil && int(st.Sys().(*syscall.Stat_t).Uid) == os.Geteuid() && st.Mode().Perm() == 0700 {
```

Comment:
This will often fail silently because (at least IME on desktop,) the default permission is `0755`.

Reply:
bumped to 755, @giuseppe any issues there?  I also bumped the MkDirall to 755 at line78